src/schema/users.ts

```diff
@@ -1044,6 +1078,55 @@ export const resolvers: IResolvers<any, Context> = {
 
       return { _: true };
     },
+    acceptFeatureInvite: async (
+      _,
+      args: {
+        token: string;
+        referrerId: string;
+        feature: string;
+      },
+      ctx: Context,
+    ): Promise<unknown> => {
+      const featureRepo = ctx.con.getRepository(Feature);
+      const existingFeature = await featureRepo.findOneBy({
+        userId: ctx.userId,
+        feature: args.feature as FeatureType,
+      });
+
+      // silently fail if we already have the feature
+      if (existingFeature?.value === FeatureValue.Allow) return undefined;
```

Comment:
If there's a row in the table, we can immediately return, no need to check if it's allowed. The only other case is if we deliberately blocked someone and in this case we don't want to grant them access

Reply:
Giving it another thought, maybe even fetching from the feature table is not required. If it already exists when trying to insert the db will return a duplicate entry error and we can catch and silently return as you suggested. It will turn this endpoint to two queries max

Reply:
I went with your suggestion, this definitely simplifies things if we don't care about existing blocked features. However the way to check for a uniqueness error on an insert is not very intuitive, and I am not sure it is the right approach:
https://github.com/dailydotdev/daily-api/blob/1add87fa682f7a9fc9a5459ae1219ad46c9ce1ba/src/schema/users.ts#L1114

Reply:
Yap, this is how we do it. We also have an enum for it, you can check the codebase for some examples. Performance wise it's better than two queries